rectools/models/sasrec.py

```diff
@@ -561,13 +540,57 @@ def get_dataloader_recommend(self, dataset: Dataset) -> DataLoader:
         return recommend_dataloader
 
 
+# ####  --------------  Lightning Model  --------------  #### #
+
+
+class SessionEncoderLightningModule(SessionEncoderLightningModuleBase):
+    """TODO"""
+
+    def on_train_start(self) -> None:
+        """TODO"""
+        self._xavier_normal_init()
+
+    def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
+        """TODO"""
+        x, y, w = batch
+        logits = self.forward(x)  # [batch_size, session_maxlen, n_items + 1]
+        # We are using CrossEntropyLoss with a multi-dimensional case
+
+        # Logits must be passed in form of [batch_size, n_items + 1, session_maxlen],
+        #  where n_items + 1 is number of classes
+
+        # Target label indexes must be passed in a form of [batch_size, session_maxlen]
+        # (`0` index for "PAD" ix excluded from loss)
+
+        # Loss output will have a shape of [batch_size, session_maxlen]
+        # and will have zeros for every `0` target label
+        if self.loss == "softmax":
+            # ce_criterion = torch.nn.CrossEntropyLoss()
+            # https://github.com/NVIDIA/pix2pixHD/issues/9 how could an old bug appear again...
+            loss_func = nn.CrossEntropyLoss(ignore_index=0, reduction="none")
```

Comment:
Since we are using `loss_func` only to calculate loss can we simple use `torch.nn.functional.cross_entropy`?

Reply:
fixed